code/modules/research/stock_parts.dm

```diff
@@ -20,9 +20,17 @@ If you create T5+ please take a pass at gene_modder.dm [L40]. Max_values MUST fi
 	var/works_from_distance = 0
 	var/pshoom_or_beepboopblorpzingshadashwoosh = 'sound/items/rped.ogg'
 	var/alt_sound = null
+	var/static/list/can_go_in = typecacheof(list(
+		/obj/item/reagent_containers/glass/beaker,
+		/obj/item/device/assembly/igniter,
+		/obj/item/stock_parts,
+		/obj/item/stack/ore/bluespace_crystal))
 
 /obj/item/storage/part_replacer/can_be_inserted(obj/item/W, stop_messages = 0, mob/user)
-	return ..() && W.get_part_rating()
+	var/yes = can_go_in[W.type]
+	if(yes)
+		return ..() && TRUE
```

Comment:
??

Reply:
see the parent proc. https://github.com/tgstation/tgstation/blob/master/code/game/objects/items/storage/storage.dm#L272